### PR TITLE
Log Out

### PR DESCRIPTION
**Pull request by mikeger**
- Updated the design for logging out
- Added confirmation
- Added deletion of the database
- Removed the feature flag for logout

# Dependencies

- [x] https://github.com/wireapp/wire-ios-sync-engine/pull/512

### CODE REVIEW

File: Wire-iOS/Sources/Helpers/syncengine/SessionManager+Shared.swift [hunk 1]
```diff
@@ -26,4 +26,26 @@ extension SessionManager {
...
+        }
+        
+        guard let sharedContainerURL = Bundle.main.appGroupIdentifier.map(FileManager.sharedContainerDirectory) else {
+            preconditionFailure("Unable to get shared container URL")
+        }
+        
+        let selectedAccountID = selectedAccount.userIdentifier
+        
+        self.logoutCurrentSession(deleteCookie: true)
+        self.accountManager.remove(selectedAccount)
```
Comment (typfel) [inline Wire-iOS/Sources/Helpers/syncengine/SessionManager+Shared.swift]: Is this safe to delete the account at this point? Maybe we should delete it inside SessionManager after it has torn down the user session.

Comment (mikeger) [reply Wire-iOS/Sources/Helpers/syncengine/SessionManager+Shared.swift]: I was under impression that `SessionManager.logoutCurrentSession` is synchronously waiting for the teardown to be done: it is calling `ZMUserSession.closeAndDeleteCookie` which is calling the operation loop teardown.

Comment (typfel) [reply Wire-iOS/Sources/Helpers/syncengine/SessionManager+Shared.swift]: You're right looks like it's blocking. I'd still argue removing the account and deleting the folder is something we should do inside SE.

Comment (typfel) [reply Wire-iOS/Sources/Helpers/syncengine/SessionManager+Shared.swift]: Maybe add a `deleteAccount:` argument to `logoutCurrentSession()`

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift [hunk 2]
```diff
@@ -223,11 +219,36 @@ extension SettingsCellDescriptorFactory {
...
+        let logoutAction: ()->() = {
+            guard let selectedAccount = SessionManager.shared?.accountManager.selectedAccount else {
+                fatal("No session manager and selected account to log out from")
+            }
+            
+            SessionManager.shared?.logoutCurrentSession(deleteCookie: true)
+            SessionManager.shared?.delete(account: selectedAccount)
+            
+            if let otherAccount = SessionManager.shared?.accountManager.accounts.first {
+                SessionManager.shared?.accountManager.select(otherAccount)
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift]: Have you tried this scenario? Would it briefly show the registration before logging into the second account?

Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift]: It's semi-working, it shows login window with X button, when you press X you land in the other account, checking now what's the best way to fix it.

Comment (typfel) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift]: Miss read, thought you were calling `select()` on the session manager. So then it's working no? Or what behaviour do you want?

Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift]: I wanted to skip the login screen, but may be it's OK like that for the moment. Will check with Yorgos.

Comment (typfel) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift]: If you want to skip the login I think we would need to have two paths. 

1. If the user has as second account switch to it and *then* delete the first account. 
2. if the user has only one account logout and delete the account.

Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift]: Actually let's merge this one so I can show it to Yorgos :)

**Pull request merged**